File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private TreeNode inOrder(TreeNode root){

        if(root == null){
            return null;
        }

        TreeNode left = inOrder(root.left);
        TreeNode right = inOrder(root.right);


            TreeNode temp = left;
            root.left = right;
            root.right = temp;
        
        return root; 
    }
    public TreeNode invertTree(TreeNode root) {
        inOrder(root);

        return root;
    }
}

File: 0226-invert-binary-tree/0226-invert-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    

    fun invertTree(root: TreeNode?): TreeNode? {
        if(root == null){
            return root
        }

        var leftNode = invertTree(root.left)
        var rightNode = invertTree(root.right)

        val temp = leftNode
        root.left = rightNode
        root.right = temp

        return root
    }
}